src/test/acquisition-sdk.ts

```diff
@@ -226,6 +226,53 @@ describe("Acquisition SDK", () => {
             done();
         }));
     });
+
+    it("doesnt disable api calls on successful response", (done: Mocha.Done): void => {
+        var acquisition = new acquisitionSdk.AcquisitionManager(new mockApi.HttpRequester(), configuration);
+
+        acquisition.queryUpdateWithCurrentPackage(templateCurrentPackage, (error: Error, returnPackage: acquisitionSdk.RemotePackage | acquisitionSdk.NativeUpdateNotification) => {
+            assert.strictEqual(acquisitionSdk.AcquisitionManager.apiCallsDisabled, false);
+        });
+
+        acquisition.queryUpdateWithCurrentPackage(templateCurrentPackage, (error: Error, returnPackage: acquisitionSdk.RemotePackage | acquisitionSdk.NativeUpdateNotification) => {
+            assert.notStrictEqual(returnPackage, null)
+        });
+
+        acquisition.reportStatusDeploy(templateCurrentPackage, acquisitionSdk.AcquisitionStatus.DeploymentSucceeded, "1.5.0", mockApi.validDeploymentKey, ((error: Error, parameter: void): void => {
+            assert.strictEqual(acquisitionSdk.AcquisitionManager.apiCallsDisabled, false);
+        }))
+
+        acquisition.reportStatusDownload(templateCurrentPackage, ((error: Error, parameter: void): void => {
+            assert.strictEqual(acquisitionSdk.AcquisitionManager.apiCallsDisabled, false);
+        }));
+        done();
+    })
+
+    it("disables api calls on unsuccessful response", (done: Mocha.Done): void => {
+        var invalidJsonResponse: acquisitionSdk.Http.Response = {
+            statusCode: 404,
+            body: "Not found"
+        };
+        configuration = { ...configuration, serverUrl: "https://codepush.appcenter.ms" }
+        var acquisition = new acquisitionSdk.AcquisitionManager(new mockApi.CustomResponseHttpRequester(invalidJsonResponse), configuration);
+
+        acquisition.queryUpdateWithCurrentPackage(templateCurrentPackage, (error: Error, returnPackage: acquisitionSdk.RemotePackage | acquisitionSdk.NativeUpdateNotification) => {
+            assert.strictEqual(acquisitionSdk.AcquisitionManager.apiCallsDisabled, true);
+        });
+
+        acquisition.reportStatusDeploy(templateCurrentPackage, acquisitionSdk.AcquisitionStatus.DeploymentSucceeded, "1.5.0", mockApi.validDeploymentKey, ((error: Error, parameter: void): void => {
```

Comment:
before calling this method `apiCallsDisabled` is already `true` as a result of previous method invocation. We need to refresh the context before testing this and the next method.